openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -202,8 +205,12 @@ def celebrations(self):
         """
         Returns a list of celebrations that should be performed.
         """
+        should_celebrate_streak = False
+        if not is_masquerading_as_specific_student(self.effective_user, self.course_key):
+            should_celebrate_streak = UserCelebration.perform_streak_updates(self.effective_user, self.course_key)
         return {
             'first_section': CourseEnrollmentCelebration.should_celebrate_first_section(self.enrollment_object),
+            'should_celebrate_streak': should_celebrate_streak,
```

Comment:
See my comments above in the other API file for comments on masquerade check and supplying the number streak being celebrated.